File: C++/Manual 09/Question04.cpp
#include <iostream>
using namespace std;

struct Division
{
    double firstQuarterSales;
    double secondQuarterSales;
    double thirdQuarterSales;
    double fourthQuarterSales;

    void Input()
    {
        cout << "==================================================================================================\n\n";
        cout << "Enter First Quarter Sales Figure: $";
        cin >> firstQuarterSales;
        cout << "Enter Second Quarter Sales Figure: $";
        cin >> secondQuarterSales;
        cout << "Enter Third Quarter Sales Figure: $";
        cin >> thirdQuarterSales;
        cout << "Enter Fourth Quarter Sales Figure: $";
        cin >> fourthQuarterSales;

        if (firstQuarterSales < 0 || secondQuarterSales < 0 || thirdQuarterSales < 0 || fourthQuarterSales < 0)
        {
            cout << "\n\nInvalid Input(s)!\n\n";
        }
        else
        {
            double totalSales = firstQuarterSales + secondQuarterSales + thirdQuarterSales + fourthQuarterSales;
            double averageSales = totalSales / 4;
            cout << "\n\n==================================================================================================\n\n";
            cout << "Annual Sales: $" << totalSales << endl;
            cout << "Average Quarterly Sales: $" << averageSales << endl;
            cout << "\n\n==================================================================================================\n\n";
        }
    }
};

int main()
{
    Division North, East, West, South;

    cout << "\t\t\t\t\tNorthern Division\n";
    North.Input();

    cout << "\t\t\t\t\tEastern Division\n";
    East.Input();

    cout << "\t\t\t\t\tWestern Division\n";
    West.Input();

    cout << "\t\t\t\t\tSouthern Division\n";
    South.Input();

    return 0;
}

File: Python/Manual 09/Question04.py
class DivisionSales:
    def __init__(self, name):
        self.name = name
        self.first_quarter = 0.0
        self.second_quarter = 0.0
        self.third_quarter = 0.0
        self.fourth_quarter = 0.0
        self.total_sales = 0.0
        self.average_sales = 0.0

    def calculate_totals(self):
        self.total_sales = (
            self.first_quarter
            + self.second_quarter
            + self.third_quarter
            + self.fourth_quarter
        )
        self.average_sales = self.total_sales / 4.0

    def set_sales(self):
        self.first_quarter = self.get_valid_sales("First Quarter")
        self.second_quarter = self.get_valid_sales("Second Quarter")
        self.third_quarter = self.get_valid_sales("Third Quarter")
        self.fourth_quarter = self.get_valid_sales("Fourth Quarter")
        self.calculate_totals()

    def get_valid_sales(self, quarter):
        while True:
            try:
                sales = float(
                    input(f"Enter {quarter} sales for {self.name} Division: ")
                )
                if sales < 0:
                    raise ValueError("Sales figures cannot be negative.")
                return sales
            except ValueError as e:
                print(e)


def main():
    divisions = []
    for name in ["East", "West", "North", "South"]:
        division = DivisionSales(name)
        division.set_sales()
        divisions.append(division)

    print("\nDivision Sales Summary")
    print("-" * 30)
    for division in divisions:
        print(f"Division: {division.name}")
        print(f"  Total Sales: ${division.total_sales:.2f}")
        print(f"  Average Quarterly Sales: ${division.average_sales:.2f}")
        print()


if __name__ == "__main__":
    main()
